api/lib/mno_enterprise/concerns/controllers/jpi/v1/app_instances_controller.rb

```diff
@@ -13,31 +13,34 @@ module MnoEnterprise::Concerns::Controllers::Jpi::V1::AppInstancesController
   #==================================================================
   # Instance methods
   #==================================================================
-  # GET /mnoe/jpi/v1/organization/1/apps.json?timestamp=151452452345
+  # GET /mnoe/jpi/v1/organization/1/apps.json?tg=151452452345
   def index
-    @app_instances = parent_organization.app_instances.active.where("updated_at.gt" => Time.at(timestamp)).select do |i|
-      # force owner assignment to avoid a refetch in ability can?(:access,i)
-      i.owner = parent_organization
-      can?(:access,i)
+    ts = Time.at(timestamp)
+    @app_instances = Mno::AppInstance.includes(:owner,:app).where(owner_id: parent_organization.id, status: Mno::AppInstance::ACTIVE_STATUSES).select do |i|
+      can?(:access,i) && i.updated_at > ts
     end
   end
 
   # POST /mnoe/jpi/v1/organization/1/app_instances
   def create
     authorize! :manage_app_instances, parent_organization
-    app_instance = parent_organization.app_instances.create(product: params[:nid])
-    MnoEnterprise::EventLogger.info('app_add', current_user.id, 'App added', app_instance)
-    head :created
+    app_instance = parent_organization.provision_app_instance(params[:nid])
+    if app_instance.errors.any?
+      render json: app_instance.errors, status: :bad_request
+    else
+      MnoEnterprise::EventLogger.info('app_add', current_user.id, 'App added', app_instance.first)
+      head :created
+    end
   end
 
   # DELETE /mnoe/jpi/v1/app_instances/1
   def destroy
-    app_instance = MnoEnterprise::AppInstance.find(params[:id])
+    app_instance = Mno::AppInstance.find(params[:id])[0]
```

Comment:
`find` is always returning an array? Unlike AR/Her which distinguish `find` from `find_all`?
In this case, I'd prefer `first` than `[0]`

Or should we use `find_one`?